From a865ae02b3aaab03dbbb7ba509b2412e14d48ae0 Mon Sep 17 00:00:00 2001
From: Fakeman <55951275+fakemanoan@users.noreply.github.com>
Date: Sun, 23 Jul 2023 14:40:05 +0100
Subject: [PATCH 2/3] Add memfd backport back

we are on kernel 3.10 (earlier than 3.18 check in code) so we need this option to have stuff function properly

Change-Id: If963d62ae6288792e51061458f292309df92b039
---
 build/soong/Android.bp     | 17 +++++++++++++++++
 config/BoardConfigSoong.mk |  2 ++
 2 files changed, 19 insertions(+)

diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index 201ab571..41107e78 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -204,6 +204,23 @@ needs_netd_direct_connect_rule {
     },
 }
 
+soong_config_module_type {
+	name: "has_memfd_backport",
+	module_type: "cc_defaults",
+	config_namespace: "lineageGlobalVars",
+	bool_variables: ["has_memfd_backport"],
+	properties: ["cppflags"],
+}
+
+ has_memfd_backport {
+	name: "has_memfd_backport_defaults",
+	soong_config_variables: {
+		has_memfd_backport: {
+    		cppflags: ["-DHAS_MEMFD_BACKPORT"],
+		},
+	},
+}
+
 soong_config_module_type {
     name: "bootloader_message_offset",
     module_type: "cc_defaults",
diff --git a/config/BoardConfigSoong.mk b/config/BoardConfigSoong.mk
index 8cb33762..7d7b8e0c 100644
--- a/config/BoardConfigSoong.mk
+++ b/config/BoardConfigSoong.mk
@@ -33,6 +33,7 @@ SOONG_CONFIG_lineageGlobalVars += \
     bootloader_message_offset \
     gralloc_handle_has_reserved_size \
     needs_netd_direct_connect_rule \
+    has_memfd_backport \
     target_health_charging_control_charging_path \
     target_health_charging_control_charging_enabled \
     target_health_charging_control_charging_disabled \
@@ -66,6 +67,7 @@ endif
 # Soong bool variables
 SOONG_CONFIG_lineageGlobalVars_gralloc_handle_has_reserved_size := $(TARGET_GRALLOC_HANDLE_HAS_RESERVED_SIZE)
 SOONG_CONFIG_lineageGlobalVars_needs_netd_direct_connect_rule := $(TARGET_NEEDS_NETD_DIRECT_CONNECT_RULE)
+SOONG_CONFIG_lineageGlobalVars_has_memfd_backport := $(TARGET_HAS_MEMFD_BACKPORT)
 SOONG_CONFIG_lineageGlobalVars_uses_egl_display_array := $(TARGET_USES_EGL_DISPLAY_ARRAY)
 SOONG_CONFIG_lineageNvidiaVars_uses_nvidia_enhancements := $(NV_ANDROID_FRAMEWORK_ENHANCEMENTS)
 SOONG_CONFIG_lineageQcomVars_supports_extended_compress_format := $(AUDIO_FEATURE_ENABLED_EXTENDED_COMPRESS_FORMAT)
-- 
2.25.1

